packages/create-bison-app/template/server/trpc.ts

```diff
@@ -0,0 +1,35 @@
+import { initTRPC, TRPCError } from '@trpc/server';
+import { TRPC_ERROR_CODE_KEY } from '@trpc/server/rpc';
+
+import type { Context } from './context';
+
+import { transformer } from '@/lib/trpc';
+
+export const t = initTRPC.context<Context>().create({
+  transformer,
+  errorFormatter({ shape, error }) {
+    return {
+      ...shape,
+      data: {
+        ...shape.data,
+        invalidArgs: error instanceof BisonError ? error.invalidArgs : undefined,
+        message: error.message,
+      },
+    };
+  },
+});
+
+export class BisonError extends TRPCError {
```

Comment:
This special error class makes it possible for errors sent to the client to include metadata. The `invalidArgs` comes from the GraphQL extension we were using for this same behavior. Other fields could be added if we wanted to extend it further.